### DISCUSSION TITLE
Looking for ideas to improve the API!

### DISCUSSION THREAD

**Discussion by josmithua** [stashenergy/react-native-msal#107] [Ideas]
Hi all that use `react-native-msal`,

First of all thanks for using this library! Hopefully it's by choice and not by necessity...

I am looking for ideas on how to improve the `react-native-msal` public facing API to make it easier and/or nicer to work with.

One of my ideas is introducing hooks, inspired by [@azure/msal-react](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/hooks.md)

But I would love to get the community's feedback on the current API and any ideas how to improve it!

Cheers,
Joshua

**Comment by k-mehta** [comment]
Tunneling exceptions would be a good addition. Here's the documented list for Android and iOS:
https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-android-handling-exceptions
https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-error-handling-ios
